database/migrations/community/2024_12_16_000002_update_forumtopic_table.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\SQLiteConnection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table('ForumTopic', function (Blueprint $table) {
+            if (DB::connection() instanceof SQLiteConnection) {
+                $table->dropForeign(['ForumID']);
+            } else {
+                $table->dropForeign('forum_topics_forumid_foreign');
```

Comment:
this failed for me:
```
QLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP FOREIGN KEY `forum_topics_forumid_foreign`; check that it exists (Connection: mysql, SQL: alter table `ForumTopic` drop foreign key `forum_topics_forumid_foreign`)
```
I don't see this foreign key in prod either:
```
  PRIMARY KEY (`ID`),
  KEY `forum_topics_forum_id_index` (`ForumID`),
  KEY `forum_topics_created_at_index` (`DateCreated`),
  KEY `forumtopic_author_id_foreign` (`author_id`),
  KEY `idx_permissions_deleted_latest` (`RequiredPermissions`,`deleted_at`,`LatestCommentID`),
  CONSTRAINT `forumtopic_author_id_foreign` FOREIGN KEY (`author_id`) REFERENCES `UserAccounts` (`ID
```

Reply:
I think my tables got in a weird state after migrating and rolling back a bunch of times. This should be fixed in latest.